protocol/avdl/keybase1/simple_fs.avdl

```diff
@@ -0,0 +1,167 @@
+@namespace("keybase.1")
+
+protocol GuiFileAccess {
+
+  import idl "common.avdl";
+
+/*
+  File manipulation interface
+  
+  Caller provides OpID to one of these methods to begin an operation:
+    list
+    listRecursive
+    copy
+    open
+    move
+    remove
+  then calls one of the following until until status is no longer pending
+  or operation is cancelled:
+    readList (after list or listRecursive)
+    read (after open)
+    write (after open)
+    check (after copy, move or remove)
+  Caller can optionally block by calling wait()
+  Operation must be closed by calling close
+ 
+*/
+
+  // Random GUID provided by caller 
+  fixed OpID(16);
+  
+  enum ItemType {
+    FILE,
+    DIR,
+    SYM,
+    EXEC
+  }
+
+  record Item {
```

Comment:
Does this need to be a `variant` since presumably the different items types will have fields that don't make sense for the others?